### ISSUE TITLE
Allow Parallelization Core Limits

### ISSUE THREAD

**Issue by adamzimmermann** [consolidation/robo#1123]
### Steps to reproduce
When you run a command with parallelization it uses all of the CPU cores, which puts a higher than wanted load on our QA server that is already running builds for multiple environments in parallel at the Docker level (outside the scope of Robo).

### Expected behavior
When you run a command with parallelization you can specify how many cores you want to use.

### Actual behavior
I am unable to specify the number of cores I want to use.

### System Configuration
This is all likely N/A to this issue, but I'm providing it anyways

- A multi-core machine from https://www.tugboatqa.com/.
- Using the [chromatichq/php-apache:8.1](https://hub.docker.com/layers/chromatichq/php-apache/8.1/images/sha256-abea9468196780a8fb4305bdf15372e1012f222f5dde54e8e79a33a94b22eb7d?context=explore) image.

### Proposed Change

I assume such a change would be done in [`ParallelExec::run`](https://github.com/consolidation/Robo/blob/4.x/src/Task/Base/ParallelExec.php#L157), presumably via a new property on the class with a setter method?

- Has this been discussed before? (I couldn't find anything in the issue queue).
- Would you be open to a PR that adds this functionality
- Are there any gotchas or wisdom that you could share before I try to build something?

**Comment by greg-1-anderson** [comment]
Yes, a PR would be welcome. I am not sure off the top of my head how resource limitations can be specified in ParallelExec. Adding setters to tasks is straightforward, though, so as long as you have a mechanism to enforce the limits you want, this should be straightforward to do.

**Comment by adamzimmermann** [comment]
I'll give this a shot. Not sure when, but it's on the todo list.
